core/src/main/scala/org/apache/spark/rdd/ParallelCollectionRDD.scala

```diff
@@ -135,10 +135,10 @@ private object ParallelCollectionRDD {
             new Range.Inclusive(r.start + start * r.step, r.end, r.step)
           }
           else {
-            new Range(r.start + start * r.step, r.start + end * r.step, r.step)
+            new Range.Inclusive(r.start + start * r.step, r.start + end * r.step - 1, r.step)
```

Comment:
What about `Range.Exclusive`?

```suggestion
            new Range.Exclusive(r.start + start * r.step, r.start + end * r.step, r.step)
```

Reply:
Range.Exclusive doesn't exist in 2.12, and Range() (exclusive in 2.12) doesn't exist in 2.13. :( I tried that initially. Because these are integers, I think we can get away with an Inclusive range that ends at end - 1 instead.

Reply:
I see. In this case this is totally fine.

Reply:
One more question: what about using `until` and `by`?

Reply:
It's probably equivalent, yeah. I was shooting for minimal change here, but there may be several solutions.